### PR TITLE
feat(tos): Show tos before login if they change to existing users

### PR DESCRIPTION
**Pull request by michalszorad**
Resolves https://emurgo.atlassian.net/browse/YOMO-700
Resolves https://emurgo.atlassian.net/browse/YOMO-692
Resolves https://emurgo.atlassian.net/browse/YOMO-688
Resolves https://emurgo.atlassian.net/browse/YOMO-685

https://github.com/Emurgo/yoroi/assets/19684576/15a09888-c4f7-4f9c-831f-201bdac54d2d

Base branch is https://github.com/Emurgo/yoroi/pull/2595/files

### CODE REVIEW

**Comment by michalszorad** [comment]
Still WIP

File: apps/wallet-mobile/src/AppNavigator.tsx [hunk 1]
```diff
@@ -79,6 +86,26 @@ export const AppNavigator = () => {
...
+                  options={{...defaultStackNavigationOptionsV2, headerShown: true, title: strings.acceptTermsTitle}}
+                />
+              </>
+            )}
+
+            {authAction === 'analytics-agreement-changed-notice' && (
+              <>
+                <Stack.Screen name="analytics-agreement-changed-notice" component={AnalyticsChangedScreen} />
+
+                {/*  TODO: Add screen to read the new analytics agreement */}
```
Comment (michalszorad) [inline apps/wallet-mobile/src/AppNavigator.tsx:105]: Do we have the analytics agreement content?

File: apps/wallet-mobile/src/components/Analytics/Analytics.tsx [hunk 2]
```diff
@@ -11,25 +11,26 @@ import {AnalyticsImage} from './AnalyticsImage'
 type Props = {
   type: 'notice' | 'settings'
   onClose?: () => void
+  onReadMore?: () => void
 }
 
-export const Analytics = ({type, onClose}: Props) => {
+export const Analytics = ({type, onClose, onReadMore}: Props) => {
```
Comment (michalszorad) [inline apps/wallet-mobile/src/components/Analytics/Analytics.tsx:17]: To allow parent redirect to the analytics screen

File: apps/wallet-mobile/src/components/BlueCheckbox/BlueCheckbox.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+import {Spacer} from '../Spacer'
+
+export type BlueCheckboxProps = {
+  checked: boolean
+  onPress?: () => void
+  children?: ReactNode
+}
+
+export const BlueCheckbox = ({checked, onPress, children}: BlueCheckboxProps) => {
```
Comment (michalszorad) [inline apps/wallet-mobile/src/components/BlueCheckbox/BlueCheckbox.tsx:14]: Extracted this checkbox into a component

File: apps/wallet-mobile/src/features/Initialization/AnalyticsChangedScreen/AnalyticsChangedScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import * as React from 'react'
+import {StyleSheet} from 'react-native'
+import {SafeAreaView} from 'react-native-safe-area-context'
+
+import {Analytics} from '../../../components'
+
+export const AnalyticsChangedScreen = () => {
+  const onReadMore = () => {
+    // TODO: Add agreement link
```
Comment (michalszorad) [inline apps/wallet-mobile/src/features/Initialization/AnalyticsChangedScreen/AnalyticsChangedScreen.tsx:9]: Do we have the analytics agreement content?

File: apps/wallet-mobile/src/features/Initialization/TermsOfServiceChangedScreen/TermsOfServiceChangedScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+import {BlueCheckbox} from '../../../components/BlueCheckbox'
+import {COLORS} from '../../../theme'
+import {useAgreeWithTermsOfService,useNavigateTo} from '../common'
+
+export const TermsOfServiceChangedScreen = () => {
+  const [tosAccepted, setTosAccepted] = React.useState(false)
+  const {agree} = useAgreeWithTermsOfService()
+
+  const onPressContinue = () => {
+    agree()
```
Comment (michalszorad) [inline apps/wallet-mobile/src/features/Initialization/TermsOfServiceChangedScreen/TermsOfServiceChangedScreen.tsx:16]: We update storage only when the user clicks on continue

File: apps/wallet-mobile/src/features/Initialization/common/terms.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+export const useAgreeWithTermsOfService = () => {
+  const storage = useStorage()
+  const queryClient = useQueryClient()
+
+  const mutationFn = useCallback(async () => {
+    const version = CONFIG.LATEST_TERMS_AND_CONDITIONS_VERSION
+    const date = new Date().toISOString()
+    await storage.join('appSettings/').setItem<TermsOfServiceAgreement>(queryKey, {version, dateAccepted: date})
```
Comment (michalszorad) [inline apps/wallet-mobile/src/features/Initialization/common/terms.ts:32]: We store the version and date when accepted

File: apps/wallet-mobile/src/AppNavigator.tsx (hunk 1 above)
Comment (stackchain) [reply apps/wallet-mobile/src/AppNavigator.tsx:105]: @MichalSzorad Leon is returning tomorrow, he will be updating the PP.

File: apps/wallet-mobile/src/AppNavigator.tsx [hunk 7]
```diff
@@ -201,17 +233,36 @@ const useHideScreenInAppSwitcher = () => {
   }, [])
 }
 
-type AuthAction = 'auth-with-pin' | 'auth-with-os' | 'request-new-pin' | 'first-run'
-const getAuthAction = (authOsEnabled: boolean, authSetting: AuthSetting): AuthAction => {
+type AuthAction =
```
Comment (stackchain) [inline apps/wallet-mobile/src/AppNavigator.tsx]: I believe this is not anymore about the authentication only Legal+Auth
```suggestion
type FirstAction =
```

File: apps/wallet-mobile/src/AppNavigator.tsx [hunk 8]
```diff
@@ -201,17 +233,36 @@ const useHideScreenInAppSwitcher = () => {
...
+  | 'auth-with-pin'
+  | 'auth-with-os'
+  | 'request-new-pin'
+  | 'first-run'
+  | 'show-terms-of-service-changed-notice'
+  | 'show-analytics-agreement-changed-notice'
+const getAuthAction = (
```
Comment (stackchain) [inline apps/wallet-mobile/src/AppNavigator.tsx]:
```suggestion
const getFirstAction = (
```

File: apps/wallet-mobile/src/AppNavigator.tsx [hunk 9]
```diff
@@ -201,17 +233,36 @@ const useHideScreenInAppSwitcher = () => {
...
+  if (isString(authSetting) && !hasAcceptedAnalytics) return 'show-analytics-agreement-changed-notice'
+
   if (authSetting === 'pin') return 'auth-with-pin'
   if (authSetting === 'os' && authOsEnabled) return 'auth-with-os'
   if (authSetting === 'os' && !authOsEnabled) return 'request-new-pin'
+
   return 'first-run' // setup not completed
 }
 
 const useAuthAction = () => {
```
Comment (stackchain) [inline apps/wallet-mobile/src/AppNavigator.tsx]:
```suggestion
const useFirstAction = () => {
```

File: apps/wallet-mobile/src/AppNavigator.tsx [hunk 10]
```diff
@@ -201,17 +233,36 @@ const useHideScreenInAppSwitcher = () => {
...
   const authSetting = useAuthSetting()
   const authOsEnabled = useAuthOsEnabled()
+  const terms = useTermsOfServiceAgreement()
+  const {isConsentRequested} = useMetrics()
 
-  return getAuthAction(authOsEnabled, authSetting)
+  return getAuthAction(authOsEnabled, authSetting, terms, isConsentRequested)
```
Comment (stackchain) [inline apps/wallet-mobile/src/AppNavigator.tsx]:
```suggestion
  return getFirstAction(authOsEnabled, authSetting, terms, isConsentRequested)
```

File: apps/wallet-mobile/src/components/BlueCheckbox/BlueCheckbox.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+import {BlueCheckbox} from './BlueCheckbox'
+
+storiesOf('BlueCheckbox', module)
+  .add('Disabled', () => (
+    <BlueCheckbox checked={false}>
+      <Text>Click Me</Text>
+    </BlueCheckbox>
+  ))
+  .add('Enabled', () => (
+    <BlueCheckbox checked={true}>
```
Comment (stackchain) [inline apps/wallet-mobile/src/components/BlueCheckbox/BlueCheckbox.stories.tsx]:
```suggestion
    <BlueCheckbox checked>
```

File: apps/wallet-mobile/src/components/BlueCheckbox/BlueCheckbox.tsx [hunk 12]
```diff
@@ -0,0 +1,75 @@
+import * as React from 'react'
+import {ReactNode} from 'react'
+import {StyleSheet, TouchableOpacity, View} from 'react-native'
+import {ClipPath, Defs, G, Path, Rect, Svg} from 'react-native-svg'
+
+import {Spacer} from '../Spacer'
+
+export type BlueCheckboxProps = {
+  checked: boolean
```
Comment (stackchain) [inline apps/wallet-mobile/src/components/BlueCheckbox/BlueCheckbox.tsx]: make booleans optional 
```suggestion
  checked?: boolean
```

File: apps/wallet-mobile/src/features/Initialization/TermsOfServiceChangedScreen/TermsOfServiceChangedScreen.tsx [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+        <Text style={styles.checkboxText}>{`${strings.tosIAgreeWith} `}</Text>
+
+        <TouchableOpacity onPress={onPressLink}>
+          <Text style={[styles.checkboxText, styles.checkboxLink]}>{strings.tosAgreement}</Text>
+        </TouchableOpacity>
+      </BlueCheckbox>
+
+      <Spacer fill />
+
+      <Button title="continue" shelleyTheme disabled={!tosAccepted} onPress={onPressContinue} />
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Initialization/TermsOfServiceChangedScreen/TermsOfServiceChangedScreen.tsx:54]: All CTAs should be sticky at bottom

File: apps/wallet-mobile/src/features/Initialization/common/terms.ts [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+  return query.data
+}
+
+export const useAgreeWithTermsOfService = () => {
+  const storage = useStorage()
+  const queryClient = useQueryClient()
+
+  const mutationFn = useCallback(async () => {
+    const version = CONFIG.LATEST_TERMS_AND_CONDITIONS_VERSION
+    const date = new Date().toISOString()
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Initialization/common/terms.ts:31]: date here will be at the render time (callback) not when clicked, I think it is fine.

File: apps/wallet-mobile/src/features/Initialization/TermsOfServiceChangedScreen/TermsOfServiceChangedScreen.tsx (hunk 13 above)
Comment (banklesss) [reply apps/wallet-mobile/src/features/Initialization/TermsOfServiceChangedScreen/TermsOfServiceChangedScreen.tsx:54]: also `continue` needs to be translated

(I need to fix this in my pr too)

**Pull request merged**